### ISSUE TITLE
Error 400: Invalid value for field 'resource.sourceDisk'

### ISSUE THREAD

**Issue by danielkubat** [hashicorp/packer-plugin-googlecompute#15]
#### Overview of the Issue

When trying to use `googlecompute, v0.0.2` plugin, packer is failing to create an image after successful build phase. Using `packer 1.7.2` binary without `required_plugins` block and `init` works fine as before for very same code (template), in same environment.

#### Reproduction Steps

Execution of packer templates with defined `required_plugins` block after `packer init`

### Plugin and Packer version

`packer 1.7.2`
`plugin 0.0.2`

### Simplified Packer Buildfile
```
source "googlecompute" "default" {
  disk_size             = var.disk_size
  disk_type             = var.disk_type
  image_description     = var.desc
  image_family          = var.image_family
  image_labels          = var.image_labels
  image_name            = var.image_name
  labels                = var.labels
  machine_type          = var.machine_type
  network               = var.network
  omit_external_ip      = var.omit_external_ip
  preemptible           = var.preemptible
  project_id            = var.project_id
  scopes                = var.scopes
  service_account_email = var.service_account_email
  source_image_family   = var.source_image_family
  ssh_username          = var.ssh_username
  subnetwork            = var.subnetwork
  tags                  = var.tags
  use_internal_ip       = var.use_internal_ip
  zone                  = var.zone
}
```

### Operating system and Environment details

CentOS7/8, RHEL 7/8 Google provided images

### Log Fragments and crash.log files

```
Step #1: ==> googlecompute.default: Creating image...
Step #1: ==> googlecompute.default: Error waiting for image: googleapi: Error 400: Invalid value for field 'resource.sourceDisk': 'https://compute.googleapis.com/compute/v1/project/zones/europe-west3-a/disks/packer-60815dbe-bb2b-1e0b-fd89-fb85adceef25'. The URL is malformed., invalid
Step #1: ==> googlecompute.default: Deleting disk...
Step #1:     googlecompute.default: Disk has been deleted!
Step #1: ==> googlecompute.default: Provisioning step had errors: Running the cleanup provisioner, if present...
Step #1: Build 'googlecompute.default' errored after 5 minutes 45 seconds: Error waiting for image: googleapi: Error 400: Invalid value for field 'resource.sourceDisk': 'https://compute.googleapis.com/compute/v1/project/zones/europe-west3-a/disks/packer-60815dbe-bb2b-1e0b-fd89-fb85adceef25'. The URL is malformed., invalid
Step #1: 
Step #1: ==> Wait completed after 5 minutes 45 seconds
Step #1: 
Step #1: ==> Some builds didn't complete successfully and had errors:
Step #1: --> googlecompute.default: Error waiting for image: googleapi: Error 400: Invalid value for field 'resource.sourceDisk': 'https://compute.googleapis.com/compute/v1/project/zones/europe-west3-a/disks/packer-60815dbe-bb2b-1e0b-fd89-fb85adceef25'. The URL is malformed., invalid
Step #1: 
Step #1: ==> Builds finished but no artifacts were created.
Finished Step
```

### Debug log: 
https://gist.github.com/danielkubat/99f7daf290d33ab6c81d9d5f25c90cda

**Comment by danielkubat** [comment]
According to logs it seems, that `/projects` is missing in PATH:

non-working (current) path: `https://compute.googleapis.com/compute/v1/project/zones/europe-west3-a/disks/packer-6081875a-fbda-995f-ae87-545379d7834d`

desired (real) path: `https://compute.googleapis.com/compute/v1/projects/project/zones/europe-west3-a/disks/packer-6081875a-fbda-995f-ae87-545379d7834d`

related documentation: https://cloud.google.com/compute/docs/reference/rest/v1/disks/list

**Comment by Filirom1** [comment]
Indeed the issue is here
https://github.com/hashicorp/packer-plugin-googlecompute/blob/main/builder/googlecompute/driver_gce.go

It works better with
```diff
diff --git a/builder/googlecompute/driver_gce.go b/builder/googlecompute/driver_gce.go
index 49f4a35..6e8e573 100644
--- a/builder/googlecompute/driver_gce.go
+++ b/builder/googlecompute/driver_gce.go
@@ -176,7 +176,7 @@ func (d *driverGCE) CreateImage(name, description, family, zone, disk string, im
                Labels:             image_labels,
                Licenses:           image_licenses,
                ImageEncryptionKey: image_encryption_key,
-               SourceDisk:         fmt.Sprintf("%s%s/zones/%s/disks/%s", d.service.BasePath, d.projectId, zone, disk),
+               SourceDisk:         fmt.Sprintf("%sprojects/%s/zones/%s/disks/%s", d.service.BasePath, d.projectId, zone, disk),
                SourceType:         "RAW",
                StorageLocations:   imageStorageLocations,
        }

```

**Comment by danielkubat** [comment]
merged: https://github.com/hashicorp/packer-plugin-googlecompute/pull/16

**Comment by danielkubat** [comment]
Issue reopened, because new plugin release is still not available.

**Comment by brandonjbjelland** [comment]
Tagging in @azr and @SwampDragons - any chance we can get a release shipped for this? I think anyone trying to use this plugin are currently blocked. I haven't tried the workaround yet but I gather one could compile the contents of `main` and point to that local binary.

**Comment by azr** [comment]
Hey @brandonjbjelland, this was just shipped for 1.0.0 !
